pkg/agent/apiserver/handlers/networkpolicy/handler.go

```diff
@@ -27,21 +27,28 @@ import (
 func HandleFunc(npq monitor.AgentNetworkPolicyInfoQuerier) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
 		name := r.URL.Query().Get("name")
-		groups := npq.GetNetworkPolicies()
-		var obj interface{}
-		if len(name) > 0 {
-			for _, group := range groups {
-				if group.Name == name {
-					obj = group
-					break
-				}
-			}
-			if obj == nil {
-				w.WriteHeader(http.StatusNotFound)
-				return
+		ns := r.URL.Query().Get("namespace")
+		if len(name) > 0 && len(ns) == 0 {
+			w.WriteHeader(http.StatusBadRequest)
+			return
+		}
+
+		policies := npq.GetNetworkPolicies()
+		var resp []networkingv1beta1.NetworkPolicy
+		for _, p := range policies {
+			if (len(name) == 0 || name == p.Name) && (len(ns) == 0 || ns == p.Namespace) {
+				resp = append(resp, p)
 			}
+		}
+
+		var obj interface{}
+		if len(name) > 0 && len(resp) == 0 {
+			w.WriteHeader(http.StatusNotFound)
+			return
+		} else if len(name) > 0 {
```

Comment:
nit: no need to have else. https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow
```
if len(name) > 0 && len(resp) == 0 {
	w.WriteHeader(http.StatusNotFound)
	return
}

var obj interface{}
if len(name) > 0 {
	obj = resp[0]
} else {
	obj = networkingv1beta1.NetworkPolicyList{Items: groups}
}
```

Reply:
Fix this in command_message.go.